### ISSUE TITLE
Entering to the Intersects analysis makes the list for the target layer to load even if the analysis has been run before

### ISSUE THREAD

**Issue by saleiva** [CartoDB/cartodb#9507]
The layer would appear already selected with no loading

**Comment by Jesus89** [comment]
Hi @ramiroaznar.

Could you add more information about this issue, please?

**Comment by ramiroaznar** [comment]
![intersect](https://user-images.githubusercontent.com/5215798/42153748-5393a568-7de4-11e8-82ee-74c93773799a.gif)

This will speed things up a lot, especially for users with lots of datasets.

**Comment by Jesus89** [comment]
I'm not able to reproduce the issue. What Target Layer are you using?. Could you attach the carto file?

Thanks @ramiroaznar

**Comment by ramiroaznar** [comment]
I think you were not able to replicate it because you need an account with lots of tables.

**Comment by ramiroaznar** [comment]
.carto: [Untitled Map 7 (on 2018-07-04 at 10.50.17).carto.zip](https://github.com/CartoDB/cartodb/files/2162921/Untitled.Map.7.on.2018-07-04.at.10.50.17.carto.zip)

But as I said, try to upload several files to your account.

**Comment by rubenmoya** [comment]
So, after some research, trying different things this is what I found out:

Our first approach was to load just `20` datasets and then use the search bar to make a request to load more, first I tried to use the `LazySelect` component, but we can't use it because it relays on the method `fetch` of a collection, and we don't have a collection here, then I tried to replicate the `LazySelect` behavior in the `NodeDataset` component, but we can't do that either because of the same reason.

The second approach was to show the loading spinner with the currently selected option, that way the users will know which dataset is selected while they wait, but we can't do that because at that point we just have the value, which works for tables, because it's just a name, but it doesn't work for nodes, since we just have `a0`, `a1`, etc, and the input shows more information, like if it's a source node or the table name.

To be honest I've run out of ideas. In my opinion, we should keep it as it is, and if it really becomes a huge problem we should invest more time in thinking how to retrieve that data in a better way and implement it differently.
